pkg/rke2/rke2.go

```diff
@@ -166,5 +176,151 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", defaultPodManifestPath)
```

Comment:
Consolidate with this:
https://github.com/rancher/rke2/blob/c8283cd8e9bf7e8edba69c1916d24e9316c42f33/pkg/rke2/rke2.go#L130